import asyncio
import random

async def time_consuming_task():
    print('Czasochłonne zadanie rozpoczęte...')
    try:
        for i in range(1, 6):
            await asyncio.sleep(random.randint(1,5))
            print(f'Krok {i} zakończony')
    except asyncio.CancelledError:
        print('Zadanie zostało przerwane')
        raise
async def main():
    task = asyncio.create_task(time_consuming_task())
    await asyncio.sleep(random.randint(1,3))    
    task.cancel()
    try:
        await task
    except asyncio.CancelledError:
        print('Error: Zadanie zostało przerwane!')
asyncio.run(main())

"""6. Write a Python program to create a coroutine that simulates a time-consuming task and use asyncio.CancelledError to handle task cancellation."""